### DISCUSSION TITLE
maxOptions: Value for "unlimited"

### DISCUSSION THREAD

**Discussion by ThomasLandauer** [orchidjs/tom-select#254] [Ideas]
I'm suggesting a value for `maxOptions` that doesn't limit the number at all, and always shows every `<option>` that's included in the `<select>`: Maybe `null`, or `0`, or `-1` etc.

If this is already possible, it needs to be documented at https://tom-select.js.org/docs/ :-)

**Comment by pwbzh** [comment]
It took me a while to understand that there was a limit to the number of entries !

I propose that there be no default limit. The normal behavior of a select is to display everything.

**Comment by ivanpoznyak** [comment]
not sure if you found the solution or not already, but based on implementation this option takes effect only if it is provided as a number
So providing some value (`null` didn't work for me), but something like an `''` or `false` - worked. 
```javascript
TomSelect.new({ ..., maxOptions: false, ...})
```
I still think that it would be better to have it work the other way around as @pwbzh described - no limit by default, if the option is there - apply it. And obviously document it properly

**Reply by ThomasLandauer** [reply]
I added this to the docs: https://github.com/orchidjs/tom-select/pull/313
I took `false` ( instead of `''`), cause it looks nicer to me :-)

**Comment by oyejorge** [comment]
> null didn't work for me

That seems to be a mistake, for [example](https://jsfiddle.net/tmjo14zk/).

```null``` would also be consistent with [maxItems](https://tom-select.js.org/docs/#maxitems)

**Comment by kreintjes** [comment]
@oyejorge (or somebody else) why is there such a low default limit (50) on the maxOptions value?

It seems to cause confusion that not all items are displayed, see the reactions in this thread and also https://github.com/orchidjs/tom-select/discussions/608. We recently switched from Select2 to TomSelect and our users also were confused by this, they thought lots of options were gone and didn't know they had to search. Granted just scrolling to a dropdown of 1000 items might not always be the quickest solution either, but it can give some inspiration on what to search for.

If there is a limit and it is enforced then maybe TomSelect can show a message at the end: "More options available, use the search function". That way it is clear to the user that there actually are more results.

Ps. Thanks for this nice select2 alternative without jQuery!

**Comment by itthinx** [comment]
Tom Select is a great alternative to Selectize and we have switched to it recently in Groups for WordPress. In its 4.1.0 release in preparation [we are now using maxOptions:null](https://github.com/itthinx/groups/commit/a711edcc2473ca0e434967813fd9b6066d44168a) motivated by the unexpected limit as described for example in this case by a user https://wordpress.org/support/topic/affichage-des-groupes-dan-la-vue-utilisateurs/
